sirepo/job_api.py

```diff
@@ -271,8 +271,30 @@ def _run_mode(request_content):
                 request_content.computeJid,
             )
         )
-    return request_content.pkupdate(
-        jobRunMode=j,
-        sbatchCores=m.sbatchCores,
-        sbatchHours=m.sbatchHours,
-    )
+    request_content.jobRunMode = j
+    return _validate_and_add_sbatch_fields(request_content, m)
+
+
+def _validate_and_add_sbatch_fields(request_content, compute_model):
+    def _element_of_list(field, list):
+        def _v(value):
+            assert value in list, \
+                f'value={value} not in enum={list}'
+        _validate_and_add(field, _v)
+
+    def _number(field, *types):
+        def _v(value):
+            assert type(value) in types and value > 0, \
+                f'value={value} expecting a {types} > 0'
+        _validate_and_add(field, _v)
+
+    def _validate_and_add(field, validate):
+        v = getattr(compute_model, field)
```

Comment:
`getattr` is not right. It should be compute_model.get(field).

Reply:
I wanted an AttributeError to be raised when field isn't found. Since we rarely use the `compute_model[field]` syntax I'm slightly embarrassed to say I didn't think of using it and came up with this instead...:)